homeassistant/components/fritzbox_callmonitor/sensor.py

```diff
@@ -64,12 +64,7 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
     try:
         host = socket.gethostbyname(host)
     except socket.error:
-        _LOGGER.warning(
-            "Could not resolve hostname: %s.\nTrying default address %s",
-            host,
-            DEFAULT_HOST,
-        )
-        host = DEFAULT_HOST
+        _LOGGER.error("Could not resolve hostname %s", host)
```

Comment:
`setup_platform` needs to fail if the lookup is not successful.

Reply:
Previously it was not failing if the host was not an IP; it just issued a warning, that the phonebook could not be found (and then did not resolve the names), but was being setup correctly, and the call status worked.
I did not want to change the functionality that much, just wanted to allow host names as well as IPs.
Should I change this nevertheless? (might break some installations, that do not care about the phonebook and are (wrongly?) working now).
What's the standard way to fail the setup? return false?

Reply:
I'm now aborting the setup if the hostname cannot be resolved, so that an error is logged and no entities are created.